doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -550,12 +562,166 @@ container sonic-interface {
 ......
 ```
 
+#### Example 2: Composite Keys with Same Number of Elements(NOT Allowed case)
+
+```yang
+......
+container sonic-interface {
+	container INTERFACE {
+		list INTERFACE_LIST {  // 1st list
+			key ifname;
+			leaf ifname {
+				type string;
+			}
+			// ...
+		}
+
+		list INTERFACE_ANOTHER_LIST { // Negative case
+			key ifname;
+			leaf ifname {
+				type string;
+			}
+			// ...
+		}
+	} 
+}
+......
+```
+
+#### Example 3: Pattern Constraints (Allowed case)
+
+In this example, `INTERFACE_LIST` uses a key `ifname` that must start with "Eth", while `INTERFACE_IPADDR_LIST` uses a key `ifname` that must start with "Vlan".
```

Comment:
Does the translation support it. Also do we have use case for it? Kindly paste if we have.

Reply:
- [validator is on place.](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py#L284)
-[test for translation and reverse translation](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-mgmt/tests/libyang-python-tests/test_sonic_yang.py#L324)
- [Config db used for above test](https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/tests/files/sample_config_db.json)

Yes, seems for telemetry client, no config may be added in sample_config_db.json. That's why no build failures. This can be in Guideline to add config.

Reply:
@praveen-li There exists a config for telemetry in [sample_config_db.json](https://github.com/sonic-net/sonic-buildimage/blob/cc6a9b6832368a107ca29edb6c828c84849f3dd2/src/sonic-yang-models/tests/files/sample_config_db.json#L1207). Any idea why this has NOT caused the build failure?